site/src/pages/tutorials/thrift/blog/implement-read.mdx

```diff
@@ -0,0 +1,250 @@
+  The annotations guarantee that the first blog post will be created in the `createBlogPost()` method before we try to retrieve in the `getBlogPost()` method.
+  ```java filename=BlogServiceTest.java
+  import org.junit.jupiter.api.MethodOrderer.OrderAnnotation;
+  import org.junit.jupiter.api.Order;
+  import org.junit.jupiter.api.TestMethodOrder;
+
+  @TestMethodOrder(OrderAnnotation.class) // Add this
+  class BlogServiceTest {
+    ...
+
+    @Test
+    @Order(1) // Add this
+    void createBlogPost() throws TException {
+      ...
+    }
+
+    @Test
+    @Order(2) // Add this
+    void getBlogPost() throws TException {
+      ...
+    }
+  }
+  ```
+3. Run all the test cases on your IDE or using Gradle.
+
+  Your client retrieved a blog post from the server successfully if the test is passed.
+
+## 4. Test an error case
+
+Let's try retrieving a blog post that does not exist.
+Add a test method to retrieve a blog post with an invalid ID, asserting an exception is thrown.
+
+```java filename=BlogServiceTest.java
+import static org.assertj.core.api.Assertions.catchThrowable;
+import example.armeria.blog.thrift.BlogNotFoundException;
+...
+@Test
+@Order(3)
+void getInvalidBlogPost() {
+  final BlogClient client = new BlogClient(server.httpUri(), "/thrift");
+  final Throwable exception = catchThrowable(() -> {
+    client.getBlogPost(Integer.MAX_VALUE);
+  });
+  assertThat(exception)
+    .isInstanceOf(BlogNotFoundException.class)
+    .extracting("reason")
+    .asString()
+    .isEqualTo("The blog post does not exist. ID: " + Integer.MAX_VALUE);
+}
+```
+
+Run all the test cases on your IDE or using Gradle.
+Check that you see the test is passed.
+
+## 5. Test retrieving multiple posts
+
+Finally, let's test if we can retrieve multiple posts.
+Add a test method like the following to create the second blog post and test retrieving the list of blog posts.
+
+```java filename=BlogServiceTest.java
+import java.util.List;
+...
+@Test
+@Order(4)
+void listBlogPosts() throws TException {
+  final BlogClient client = new BlogClient(server.httpUri(), "/thrift");
+  client.createBlogPost("My second blog", "Armeria is awesome!");
+
+  final List<BlogPost> blogs = client.listBlogPosts(false);
+  assertThat(blogs).hasSize(2);
+  final BlogPost firstBlog = blogs.get(0);
+  assertThat(firstBlog.getTitle()).isEqualTo("My first blog");
+  assertThat(firstBlog.getContent()).isEqualTo("Hello Armeria!");
+
+  final BlogPost secondBlog = blogs.get(1);
+  assertThat(secondBlog.getTitle()).isEqualTo("My second blog");
+  assertThat(secondBlog.getContent()).isEqualTo("Armeria is awesome!");
+  System.out.println(blogs);
+}
+```
+
+Run all the test cases on your IDE or using Gradle.
+Check that you see the test is passed.
+
+## What's next
+
+In this step, we've implemented service methods and client methods to retrieve blog posts.
+See [Decorating a client](/docs/client-decorator/) for more information.
```

Comment:
I think this can be removed since this section doesn't really talk about client decorators

```suggestion
```